### PR TITLE
Added non-@autoclosure version of init.

### LINKED ISSUE #119: Cannot use auto closure init in a closure
for example:
```swift
func(){
let result: Result<String, NSError> = Result(try tryIsSuccess("success")) //This is OK in function
}

let simpleBlock: ()->Void =  {
let result: Result<String, NSError> = Result(try tryIsSuccess("success")) //This is not OK in colure
}
simpleBlock()
```

So my suggestion is to add a init version without @autoclosure mark
```swift
extension Result{
    public init(_ f: () throws -> T) {
        self.init(try f())
    }
}
```

### PR DESCRIPTION
**Pull request by NachoSoto**
Fixes #119. Alternative approach to #120.

### CODE REVIEW

File: Result/Result.swift [hunk 1]
```diff
@@ -17,24 +17,28 @@ public enum Result<T, Error: ErrorType>: ResultType, CustomStringConvertible, Cu
...
 	}
 
-	/// Constructs a result from a function that uses `throw`, failing with `Error` if throws
+	/// Constructs a result from a function that uses `throw`, failing with `Error` if throws.
 	public init(@autoclosure _ f: () throws -> T) {
+		self.init(f: f)
+	}
+
+	/// Constructs a result from a function that uses `throw`, failing with `Error` if throws.
+	public init(@noescape f: () throws -> T) {
```
Comment (NachoSoto) [inline Result/Result.swift]: I don't think we can make this a pure overload (as in, without the named parameter) and make the other constructor unambiguously delegate to this. It's actually quite disappointing that Swift can't even infer at compile time that doing `self.init(f)` is an infinite loop.

**Comment by gfontenot** [comment]
:+1: from me

File: Result/Result.swift (hunk 1 above)
Comment (robrix) [inline Result/Result.swift]: Can we give the parameter a better name than `f` at least?

File: Result/Result.swift [hunk 2]
```diff
@@ -17,24 +17,28 @@ public enum Result<T, Error: ErrorType>: ResultType, CustomStringConvertible, Cu
...
 	}
 
-	/// Constructs a result from an Optional, failing with `Error` if `nil`
+	/// Constructs a result from an Optional, failing with `Error` if `nil`.
 	public init(_ value: T?, @autoclosure failWith: () -> Error) {
 		self = value.map(Result.Success) ?? .Failure(failWith())
 	}
 
-	/// Constructs a result from a function that uses `throw`, failing with `Error` if throws
+	/// Constructs a result from a function that uses `throw`, failing with `Error` if throws.
```
Comment (robrix) [inline Result/Result.swift]: Thank you. These have been bothering me :grin:

File: Result/Result.swift (hunk 1 above)
Comment (NachoSoto) [inline Result/Result.swift]: Sure thing. Other ideas:
- block
- function
- closure
- attempt

Comment (robrix) [inline Result/Result.swift]: I think `attempt` sounds great :+1:

Comment (NachoSoto) [inline Result/Result.swift]: Done

**Comment by robrix** [comment]
:heart: this, @NachoSoto.

**Pull request merged**